src/slam_karto.cpp

```diff
@@ -280,8 +301,41 @@ SlamKarto::SlamKarto() :
     mapper_->setParamUseResponseExpansion(use_response_expansion);
 
   // Set solver to be used in loop closure
-  solver_ = new SpaSolver();
-  mapper_->SetScanSolver(solver_);
+  std::stringstream solver_plugin_stream;
+  solver_plugin_stream << "karto_plugins::" << solver_type_;
+  try
+  {
+    solver_ = solver_loader_.createInstance(solver_plugin_stream.str());
+    ROS_INFO_STREAM("Loaded " << solver_plugin_stream.str() << " solver plugin");
+  }
+  catch(pluginlib::PluginlibException& ex)
+  {
+    ROS_ERROR_STREAM(
+        "The solver plugin " << solver_plugin_stream.str() << " failed to load for some reason. Error: " << ex.what()
+            << ". We will load the default solver plugin (SPA).");
+    solver_ = boost::shared_ptr<karto_plugins::SPASolver>(new karto_plugins::SPASolver());
```

Comment:
Note that for solvers and visualizers, I set the fallback behavior to loading the SPA solvers/visualizers. I am not convinced that this is actually better. Should we shutdown the node instead ? (outside of the constructor preferrably)